src/api/materials.py

```
@@ -0,0 +1,541 @@
+                    import js  
+                    raw = bytes(js.Uint8Array.new(arr_buf))
+                except Exception:
+                    raw = b""
+                fields[field_name] = {"filename": filename, "bytes": raw}
+            else:
+                fields[field_name] = str(val)
+
+    return fields, None
+
+
+def _r2_bucket(env):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_r2_bucket`

(ANN202)